packages/grid/x-data-grid-pro/src/utils/domUtils.ts

```diff
@@ -15,6 +15,9 @@ export function findGridCellElementsFromCol(col: HTMLElement): NodeListOf<Elemen
   if (!root) {
     throw new Error('MUI: The root element is not found.');
   }
-  const cells = root.querySelectorAll(`.${gridClasses.cell}[data-field="${field}"]`);
+  const cells = root.querySelectorAll(
+    // Include cells that have colspan > 1 and allocate `field` cell space
+    `.${gridClasses.cell}[data-field="${field}"], .${gridClasses.cell}[data-colspan-allocates-field-${field}="1"]`,
```

Comment:
Did you consider calling `getCellColSpanInfo` to get the cell at `leftVisibleCellIndex` when it's not spanned? In this way, I think we can remove these custom attributes. They work but if one column spans many columns we'll have a lot of attributes.

Reply:
That's a good idea. The problem is that here I need `field` not `index` , but I will try to do it

Reply:
Another thing I was thinking about was to create the lookup with the field, instead of the index.

Reply:
> Another thing I was thinking about was to create the lookup with the field, instead of the index.

Not needed, I've managed to use `getCellColSpanInfo` here without it.
It's great to get rid of those data attributes that I've used previously